lib/restrictions.js

```diff
@@ -86,7 +109,15 @@ export async function bulkInsert({restrictions} = {}) {
   assert.array(restrictions, 'restrictions');
   const now = Date.now();
   const meta = {created: now, updated: now};
-  let records = restrictions.map(restriction => ({restriction, meta}));
+  let records = restrictions.map(function(restriction) {
+    let id = `${restriction.zone}${restriction.resource}${restriction.method}`;
+    if(restriction.method === 'limitOverDuration' &&
+    restriction.methodOptions && restriction.methodOptions.duration) {
+      id = `${id}${restriction.methodOptions.duration}`;
+    }
+    restriction.id = crypto.createHash('sha256').update(id).digest('base64');
```

Comment:
So, unfortunately, creating a deterministic hash of the zone, resource, and method just recreates the same problem we had with the unique index. A unique hash on the combination of fields X, Y, Z + a unique index on that value is equivalent to a unique compound index on fields X, Y, Z.

We don't want to limit a zone to only one restriction of a certain method for a given resource. The ID should just be a random generated ID (use bnid) and we should index on that. With this PR, we're deciding that a zone can have many different restrictions on the same resource.

Side note: While we shouldn't keep this hash for the reasons above, I should also point out that if we were going this route, we would need delimiters between the three fields (X, Y, Z), to prevent accidental/intentional harmful overlap.

Reply:
This still needs to be removed.